### ISSUE TITLE
consistent उदात्त svara notation

### ISSUE THREAD

**Issue by vvasuki** [sanskrit-lexicon/csl-ldev#7]
Consider the below in the context of https://github.com/sanskrit-lexicon/csl-ldev/issues/6#issuecomment-1042849671 - 
```
From Bohtlingk-and-Roth-Grosses-Petersburger-Worterbuch (sa-de)Collapse article
रूप

रूप꣫
```
It's best to settle on a consistent udAtta notation (꣡) for use in devanAgarI renderings.

**Comment by vvasuki** [comment]
Also:
```
From capeller-sanskrit-german (sa-de)Collapse article
रूप

रूप॑ n. (adj.) —° f. आ, selten ई) Aussehen,
Form, Gestalt, Farbe oft adj. —ähnlich,
```

**Comment by gasyoun** [comment]
> It's best to settle on a consistent udAtta notation (꣡) for use in devanAgarI renderings.

You want to have it not as per scan?

**Comment by vvasuki** [comment]
> > It's best to settle on a consistent udAtta notation (꣡) for use in devanAgarI renderings.
> 
> You want to have it not as per scan?

Yes - change it. ꣫ has long been used in sAmaveda texts to mean something other than "udAtta" which is what the author intended to denote. It has been included in unicode expressly in the former sense. Hence it is just confusing for users and machines. A simple shift will avoid it, without any drawbacks.

**Comment by drdhaval2785** [comment]
Let us differentiate different types of changes, and discuss further.

# Type 1

There is a typographical error. Change is made in digitized version to conform to printed text. All agree that it should be corrected.

# Type 2

There is no change in content. Only display is to be changed. Glyph issue raised above falls in that category. There have been many such display changes adopted. No issues.

# Type 3

Changes in encoding or transliteration standards. Every era has its preferred transliteration scheme. Over the years CDSL has been making such changes e.g. cp1252 to unicode. Anglicised Sanskrit to IAST. Sanskrit text from different dictionaries in different transliterations brought to SLP1. No issues.

# Type 4

Changes where scholars differ, on principles. There has not been consensus about the accepted way of rendering accents across Sanskrit scholars. A Samaveda scholar may mark the same accent in a different way than Rigveda or Yajurveda. In such cases, a discussion is required to reach to a consensus. Some published papers or documentation would help. Otherwise, it may be decided based on what participants feel. Once consensus is reached, this can be taken up.

# Type 5

Changes which are based on user’s preferences of user’s gut feeling. We should discourage such changes, because all users have different feelings or preferences. These changes also tend to create inconsistencies, because the user would submit only for the item he is interested in. There may be many such cases where such a change is not effected.

**Comment by gasyoun** [comment]
> Obviously no one is imitating that style now! 

Yes, but should we care?

>Glyph issue raised above falls in that category

Not sure. Or let's add more examples in each group.

**Comment by vvasuki** [comment]
> > Some published papers or documentation would help. Otherwise, it may be decided based on what participants feel. Once consensus is reached, this can be taken up.
> 
> Sure. I've made a proposal. Let's reach a consensus next. How will you go about it?

One way of doing this is to look at unicode documentations - pdfs linked in https://en.wikipedia.org/wiki/Devanagari_(Unicode_block) , https://en.wikipedia.org/wiki/Devanagari_Extended and https://en.wikipedia.org/wiki/Vedic_Extensions . They have been produced after long and detailed discussions among topic experts. If you have a better notation in mind from there - propose it. In any case, as you said: 

> "Only display is to be changed. Glyph issue raised above falls in that category. There have been many such display changes adopted. No issues."

So any changes we make here would be easily reversible in the future.

**Comment by vvasuki** [comment]
> There are three standards mentioned in the above link. Devanagari - [U0900.pdf](https://github.com/sanskrit-lexicon/csl-ldev/files/8096517/U0900.pdf) Devanagari Extended - [UA8E0.pdf](https://github.com/sanskrit-lexicon/csl-ldev/files/8096523/UA8E0.pdf) Vedic Extensions - [U1CD0.pdf](https://github.com/sanskrit-lexicon/csl-ldev/files/8096525/U1CD0.pdf)

Minor correction - They're not really three standards - as in three **competing** standards. Just 3 code-point tables. Our choice for udAtta notation should necessarily come from them.

**Comment by drdhaval2785** [comment]
Vedic accent and lexicography - https%3A%2F%2Fzenodo.org%2Frecord%2F837826%2Ffiles%2FRau2017_vedic-accent-in-lexicography.pdf

**Comment by vvasuki** [comment]
> Vedic accent and lexicography - https://zenodo.org/record/837826/files/Rau2017_vedic-accent-in-lexicography.pdf

Good summary.

![image](https://user-images.githubusercontent.com/2664797/154714884-0c0859b8-853d-412c-bd7b-9973ba1ec7fc.png)


## Böhtlingk & Roth
> This is the rationale behind the innovation of the Böhtlingk & Roth system de-
scribed above. This system is used in the headwords of entries and marks udātta accent
by a diacritical raised उ above the syllable क꣫ and independent svarita by the vertical stroke
क॑.

Recipe for grand confusion (given https://github.com/sanskrit-lexicon/csl-ldev/issues/7#issuecomment-1043166159)- should use क꣡ and क꣢ respectively. (Creating a separate issue for the internal consistency and CSL encoding problems.)

## MW
> Accent is only marked in the Latin
script rendering of headwords and is virtually identical with the system described in ISO
15919. Udātta accent is marked by an acute accent á and independent svarita is marked
by grave accent à.

In devanAgarI -  क꣡ and क꣢ respectively. 

## Cappeller
> It uses the vertical stroke over the central glyph of the syllable
क॑ to indicate udātta accent and a lying J-shaped diacritic below the syllable क᳗ to indicate
independent svarita.

Ditto.

**Comment by Andhrabharati** [comment]
I guess the Samaveda svara notations (it being the one which heavily depending on various svaras and glides to recite the Riks) could be adopted across all vedas and their associated works like Brahmanas, Aranyakas, ...),  if a consistent 'common' notation is being considered (though it would differ with present published works-- not just the lexical ones being discussed here, sometimes slightly & manytimes to a major extent wrt each individual work).

But I highly doubt if this would get any true appreciation from "non-techies"!!

People are well accustomed to see different systems present in different works for ages now, and "no real complaint" seems to have comeup from any corner to make a "common framework"-- to the best of my knowledge.

**Comment by vvasuki** [comment]
> But I highly doubt if this would get any true appreciation from "non-techies"!!

The above is false (as clarified below). Motivation is not some techie obsession but linguistic interest.

> People are well accustomed to see different systems present in different works for ages now, and "no real complaint" seems to have comeup from any corner to make a "common framework"-- to the best of my knowledge.

That's because very few people (esp. very few Indian veda memorizers) actually care to look at dictionaries for prakRti-svara-s .   In fact, western sanskrit scholars routinely use and record those - not Indians (unfortunate but true). Those that actually care to do so (example- myself and a few friends) and use that information (esp in our speech and writing), would rather have some consistent notation across dictionaries and not have to memorize distinct dictionary conventions (not veda conventions)!

**Comment by Andhrabharati** [comment]
Good to see your explanation; so you want only the digitised dictionaries to be with common svara notations across all cited vedic literature, even if by doing so they might not be matching with the corresp. vedic texts in print.

Isn't better then that this be done in the 'stardict' (or babylon) branch of cdsl texts with a separate script in addition to format conversion (keeping the main cdsl texts as is, them being for a larger user group)?

**Comment by vvasuki** [comment]
> Good to see your explanation; so you want only the digitised dictionaries to be with common svara notations across all cited vedic literature, even if by doing so they might not be matching with the corresp. vedic texts in print.

To clarify, my main concern is with headwords. That should have consistent notation. Users want to look up prakRti-svara in a jiffy.

*vedic citations* should have internally consistent notation - whatever is right for the shAkhA being cited. Caveats 

- Stuff like https://github.com/sanskrit-lexicon/csl-ldev/issues/9#issue-1143328528 (where SB excerpts are presented in contrary notations) should ideally be fixed. 
- If rendered in a different notation (eg. western system of marking udAttas and jAtya svarita-s only), then that should be made consistent.

> 
> Isn't better then that this be done in the 'stardict' (or babylon) branch of cdsl texts with a separate script in addition to format conversion (keeping the main cdsl texts as is, them being for a larger user group)?

No - people who care to learn about svara-s from dictionaries don't always use stardict; and they will benefit from consistency. And consistent notation will anyway not bother those who don't care about svara-s.

**Comment by Andhrabharati** [comment]
> 
> *vedic citations* should have internally consistent notation - whatever is right for the shAkhA being cited. Caveats 
> 
> - Stuff like https://github.com/sanskrit-lexicon/csl-ldev/issues/9#issue-1143328528 (where SB excerpts are presented in contrary notations) should ideally be fixed. 
> - If rendered in a different notation (eg. western system of marking udAttas and jAtya svarita-s only), then that should be made consistent.
>

Yes, such inconsistencies (or errors in some cases) should definitely be corrected.

But very few people have that knowledge or capacity (and mostly, interest) to do such corrections.

> > 
> > Isn't better then that this be done in the 'stardict' (or babylon) branch of cdsl texts with a separate script in addition to format conversion (keeping the main cdsl texts as is, them being for a larger user group)?
> 
> No - people who care to learn about svara-s from dictionaries don't always use stardict; and they will benefit from consistency. And consistent notation will anyway not bother those who don't care about svara-s.

Learning svara-s through dictionaries? Sounds intesting; but a bad choice of 'proper sources', I'd say.

**Comment by vvasuki** [comment]
> 
> Learning svara-s through dictionaries? Sounds intesting; but a bad choice of 'proper sources', I'd say.

I disagree with this prejudice (not to mention the unspecificity - most ignoramuses understand "learning svara-s" to mean learning to say veda-mantra-s in their neo-traditional intonations). I find the laxya, traditional padapATha-s (which I've dictified as well) and vyAkaraNa to be quite consistent with the efforts of these lexicographers (who have meticulously gone through and in some cases - cited their sources).

**Comment by vvasuki** [comment]
@drdhaval2785 Can we please solve this just for headwords? It would be simple, if you agree on the devanAgarI notation suggested.

We've begun practicing svaras ( https://groups.google.com/g/shabda-shAstram/c/1izIRQfA5b8 ), and consistency in the dicts we refer to would help much.

**Comment by gasyoun** [comment]
> a consistent 'common' notation is being considered (though it would differ with present published work

Would it be common or the new common?

> People are well accustomed to see different systems present in different works for ages now, and "no real complaint" seems to have comeup from any corner to make a "common framework

They are not. Keeping silent does not means it's fine.

**Comment by vvasuki** [comment]
> > a consistent 'common' notation is being considered (though it would differ with present published work
> 
> Would it be common or the new common?
> 

As far as devanAgarI is concerned, the image in https://github.com/sanskrit-lexicon/csl-ldev/issues/7#issuecomment-1044738582 indicates that there was no "common" notation as far as the dicts were concerned (they were never intended to be used side by side). So, "new common".

Btw, I messaged the below to @drdhaval2785 on whatsapp:

1. उ is used in the sense different from udAtta in sAmaveda, and is not used in the sense of udAtta anywhere (in pre-modern or later vedic productions).
2. अ॑ is popularly used in the sense of svarita in multiple shAkhA-s.
3. There is no ambiguity as to  अ꣡ meaning udAtta (and it is used as such in the sAmaveda Archika texts).
4. There is no ambiguity as to क᳡ meaning a jAtya-svarita (and it is used as such in shaunaka saMhitA).

For our immediate purposes, it suffices if you leave aside point 4 and agree with points 1-3. Even there, point number 2 is critical (enough to cause one to change the devanAgarI headwords).


> > People are well accustomed to see different systems present in different works for ages now, and "no real complaint" seems to have comeup from any corner to make a "common framework
> 
> They are not. Keeping silent does not means it's fine.

Besides, it is a matter of fact that most people are not "accustomed" to looking up svaras like this side-by-side :
![image](https://user-images.githubusercontent.com/2664797/190835608-84c35090-8a37-4648-805d-ac491b9db65c.png)

**Comment by vvasuki** [comment]
> > > a consistent 'common' notation is being considered (though it would differ with present published work
> > 
> > 
> > Would it be common or the new common?
> 
> As far as devanAgarI is concerned, the image in [#7 (comment)](https://github.com/sanskrit-lexicon/csl-ldev/issues/7#issuecomment-1044738582) indicates that there was no "common" notation as far as the dicts were concerned (they were never intended to be used side by side). So, "new common".

If you're jittery about changing the original, you can keep it, as well as add the headword in standard notation. For example: त꣡र्हि   (Pr: त꣫र्हि). 

(I just needed to look up svara for उचित, and wasted time wondering what that diacritic over ta was supposed to mean.
![image](https://user-images.githubusercontent.com/2664797/190839211-f44702d7-14dd-4bf4-8a4c-da451377c440.png)
)

**Comment by gasyoun** [comment]
> wasted time wondering what that diacritic over ta was supposed to mean.

@funderburkjim guess it's a strong argument

**Comment by vvasuki** [comment]
Just noticed that in capeller, we get the headword accented as शपथ्य^ - so the jAtya-svarita symbol is not being translated appropriately. Could be shown श॒प॒थ्य᳡ or more simply - शपथ्य᳡
